File: src/alogorizm/sort/selectsort/SelectSort.java
package alogorizm.sort.selectsort;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Random;

public class SelectSort {
    public void sort(ArrayList<Integer> arrayList){
        int select;
        int size = arrayList.size();
        for (int i = 0; i < size; i++) {
            select = i;
            for (int j = i + 1; j < size ; j++) {
                if(arrayList.get(select) > arrayList.get(j)){
                    select = j;
                }
            }
            Collections.swap(arrayList, i, select);
        }
    }

    public static void main(String[] args) {
        SelectSort selectSort = new SelectSort();
        ArrayList<Integer> arrayList = new ArrayList<Integer>();
        Random random = new Random();
        for (int i = 0; i < 100; i++) {
            arrayList.add(random.nextInt(100));
        }
        selectSort.sort(arrayList);

        arrayList.forEach(System.out::println);

    }
}

File: src/alogorizm/sort/selectsort/SelectSort.kt
package alogorizm.sort.selectsort

import java.util.*
import kotlin.random.Random

fun selectSort(list: MutableList<Int>) {
    val size = list.size
    for (i in 0 until size - 1) {
        var minIdx = i
        for (j in i + 1 until size) {
            if (list[minIdx] > list[j]) {
                minIdx = j
            }
        }
        if (minIdx != i) {
            Collections.swap(list, minIdx, i)
        }
    }
}
fun main() {
    val mutableList = Array(10) { Random.nextInt(0, 100) }.toMutableList()
    println(mutableList)
    selectSort(mutableList)
    println(mutableList)
}
